### PR TITLE
Support apple_library as a dependency of swift_library

### PR DESCRIPTION
**Pull request by nguyentruongtho**
This PR will enable `apple_library` to be a (transparent) dependency of `swift_library`, yet only for `apple_library` without objc source that is referenced from library that depends on it.

### CODE REVIEW

File: test/com/facebook/buck/swift/testdata/swift_on_objc/main.m [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#import <UIKit/UIKit.h>
+#import "AppDelegate.h"
+#import <binary-Swift.h>
```
Comment (ryu2) [inline test/com/facebook/buck/swift/testdata/swift_on_objc/main.m]: Use `""` instead.

File: test/com/facebook/buck/swift/SwiftIOSBundleIntegrationTest.java [hunk 2]
```diff
@@ -279,4 +279,33 @@ public void testSwiftPreferredLinkage() throws Exception {
         workspace.runCommand("nm", binaryOutput.toString()).getStdout().or(""),
         containsString("baz"));
   }
+
+  @Test
+  public void swiftDependsOnObjCRunsAndPrintsMessageOnOSX() throws Exception {
```
Comment (ryu2) [inline test/com/facebook/buck/swift/SwiftIOSBundleIntegrationTest.java]: You don't mean `OnOSX` here.

File: test/com/facebook/buck/swift/SwiftOSXBinaryIntegrationTest.java [hunk 3]
```diff
@@ -124,4 +124,5 @@ public void swiftCallingObjCRunsAndPrintsMessageOnOSX() throws IOException {
         runResult.getStdout(),
         containsString("Hello ObjC\n"));
   }
+
```
Comment (nguyentruongtho) [inline test/com/facebook/buck/swift/SwiftOSXBinaryIntegrationTest.java]: this is a newline at the end of the file.

File: test/com/facebook/buck/swift/SwiftIOSBundleIntegrationTest.java (hunk 2 above)
Comment (nguyentruongtho) [inline test/com/facebook/buck/swift/SwiftIOSBundleIntegrationTest.java]: good catch, I copy-pasted -_-

**Comment by nguyentruongtho** [comment]
+1 for how fast facebook has moved, migrated java7->java8 within a week and requires me to rebase my PR everytime :trollface:

**Pull request closed without merging**